### PR TITLE
SDL_video: Added SDL_GL_FLOATBUFFERS to allow Cocoa GL contexts to use EDR

### PR DESCRIPTION
**Pull request by Shootfast**
This patch adds a new attribute "SDL_GL_FLOATBUFFERS" to the SDL_GLattr enum.

## Description
This allows the context creation to be passed the hint that the renderer should support float values (and pixel values >1.0).

This is required on OSX for an OpenGL context to be treated as EDR, though might also be useful on other platforms.
The OSX use case is described here:
https://developer.apple.com/videos/play/wwdc2021-10161/?time=1242

And the attribute is documented here:
https://developer.apple.com/documentation/appkit/1436213-opengl_pixel_format_attributes/nsopenglpfacolorfloat?language=objc

### CODE REVIEW

File: include/SDL_video.h [hunk 1]
```diff
@@ -237,6 +237,7 @@ typedef enum
     SDL_GL_STEREO,
     SDL_GL_MULTISAMPLEBUFFERS,
     SDL_GL_MULTISAMPLESAMPLES,
+    SDL_GL_FLOATBUFFERS,
```
Comment (sezero) [inline include/SDL_video.h:240]: You are changing the abi by inserting here instead of appending: Wrongest thing to do.

Comment (Shootfast) [reply include/SDL_video.h:240]: Ah shoot sorry, I wasn't thinking about the ABI ramifications. I'll reshuffle this.

File: src/video/x11/SDL_x11opengl.c [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+        /*
```
Comment (sezero) [inline src/video/x11/SDL_x11opengl.c:1]: Please lose the unwanted indentation here

**Comment by sezero** [comment]
OK, now the maintainers should review.

File: include/SDL_video.h (hunk 1 above)
Comment (Shootfast) [reply include/SDL_video.h]: I've moved the new entry to the end of the enum. I assume the changes to ```SDL_sysvideo.h``` are ok, as that is an internal struct?

**Comment by icculus** [comment]
Relevant extension for EGL: https://registry.khronos.org/EGL/extensions/EXT/EGL_EXT_pixel_format_float.txt

**Pull request merged**